src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -286,23 +892,73 @@ uint32 newfs_init(char *devname, char *volname, uint32 blocksize, uint32 numbloc
     if (strlen(devname) >= OS_FS_DEV_NAME_LEN || strlen(volname) >= OS_FS_VOL_NAME_LEN)
         return OS_FS_ERROR;
 
-    //the first filesystem is always the tar
-    if (!fs[0].devname) {
+
+    // the first filesystem is always the tar
+    if (!filesystems[0].devname) {
         ret = tar_load();
         if (ret != OS_FS_SUCCESS) return ret;
-        ret = fs_init(&fs[0], devname, volname, blocksize, numblocks);
+        ret = fs_init(&filesystems[0], devname, volname, blocksize, numblocks);
```

Comment:
What is the difference between `newfs_init` and `fs_init`? Some name changing might be in order